src/core/public/plugins/plugin_loader.test.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { loadPluginBundle } from './plugin_loader';
+
+let createdScriptTags = [] as any[];
+let appendChildSpy: jest.Mock<Node, [Node]>;
+let createElementSpy: jest.Mock<HTMLElement, [string, (ElementCreationOptions | undefined)?]>;
+
+beforeEach(() => {
+  // Mock document.createElement to return fake tags we can use to inspect what
+  // loadPluginBundles does.
+  createdScriptTags = [];
+  createElementSpy = jest.spyOn(document, 'createElement').mockImplementation(() => {
+    const scriptTag = { setAttribute: jest.fn() } as any;
+    createdScriptTags.push(scriptTag);
+    return scriptTag;
+  });
+
+  // Mock document.body.appendChild to avoid errors about appending objects that aren't `Node`'s
+  // and so we can verify that the script tags were added to the page.
+  appendChildSpy = jest.spyOn(document.body, 'appendChild').mockReturnValue({} as any);
+
+  // Mock global fields needed for loading modules.
+  window.__kbnNonce__ = 'asdf';
+  window.__kbnBundles__ = {};
+});
+
+afterEach(() => {
+  appendChildSpy.mockRestore();
+  createElementSpy.mockRestore();
+  delete window.__kbnNonce__;
+  delete window.__kbnBundles__;
+});
+
+const addBasePath = (path: string) => path;
+
+test('`loadPluginBundles` creates a script tag and loads initializer', async () => {
+  const loadPromise = loadPluginBundle(addBasePath, 'plugin-a');
+
+  // Verify it sets up the script tag correctly and adds it to document.body
+  expect(createdScriptTags).toHaveLength(1);
+  const fakeScriptTag = createdScriptTags[0];
+  expect(fakeScriptTag.setAttribute).toHaveBeenCalledWith(
+    'src',
+    '/bundles/plugin/plugin-a.bundle.js'
+  );
+  expect(fakeScriptTag.setAttribute).toHaveBeenCalledWith('id', 'kbn-plugin-plugin-a');
+  expect(fakeScriptTag.setAttribute).toHaveBeenCalledWith('nonce', 'asdf');
+  expect(fakeScriptTag.onload).toBeInstanceOf(Function);
+  expect(fakeScriptTag.onerror).toBeInstanceOf(Function);
+  expect(appendChildSpy).toHaveBeenCalledWith(fakeScriptTag);
```

Comment:
Is it possible to add integration test to test whole pipeline up to a browser?

Reply:
I just made some fixes to our tooling last week to allow TS in functional tests for plugins, so I think this should be possible now. I'll experiment with this, as I agree it's important.